website/blog/2024-09-01-model-as-code-logging/index.md

```diff
@@ -0,0 +1,177 @@
+
+On the tools side, MLflow continues to solidify its position as the favorite tool for MLOps among the ML community. However, the rise of GenAI has introduced new needs in how we use MLflow. One of these new challenges is how we log model artifacts in MLflow. If you’ve used MLflow before (and we bet you have), you’re probably familiar with the `mlflow.log_model()` function and how it efficiently **pickles** model artifacts (I’ve bolded "pickles" because it’s a key term in this post).
+
+Particularly with GenAI, there’s a new requirement: logging the model "as code," instead of serializing it into a pickle file. And guess what? This need isn’t limited to GenAI models. So, in this post we will explore this concept and how MLflow has adapted to meet this new requirment.
+
+You will notice that this feature is implemented at a very abstract level, allowing you to log any model "as code", whether it’s GenAI or not! We like to think of it as a generic approach, with GenAI models being just one of its use cases. So, in this post, we’ll explore this new approach, "Model-as-Code logging".
+
+By the end of this post you should be able to answer the three main questions: "what", "why" and "how" to do Model-as-Code logging.
+
+## What Is Actually Model-as-Code Logging?
+
+In fact, when MLflow announced this feature, it got us thinking in a more abstract way about the concept of a "model"! You might find it interesting, too, if you zoom out and consider a model as a mathematical representation or function that describes the relationship between input and output variables. At this level of abstraction, a model can be many things!
+
+At this level, one might recognize that a model, as an object or artifact, represents just one form of what a model can be, even if it’s the most popular in the ML community. But if you think about it, a model can also be a simple mapping function — just a piece of code — or even code that sends API requests to another service that doesn’t necessarily reside within your "premises" (e.g., OpenAI APIs).
+
+We'll explain the detailed workflow of how to log Mode-as-Code later in the post, but for now, let's consider it at a high level with two main steps: first, writing your model code, and second, logging your model as code. This will look like the folowing figure:
+
+![High Level Model-as-Code Logging Workflow](model_as_code1.png)
+
+🔴 It's important to note that when we refer to "model code," we're talking about code that can be treated as a model itself. This means it's **not** your training code that generates a trained model object, but rather the step-by-step code that is executed as a model itself.
```

Comment:
Maybe "inference code"?

Reply:
No, it is the "training code" is what the reader might confuse with the "model code" that we are talking about! ML engineers used to write code to produce model (objects) (e.g., Sklean linear regression) and they might think this is what this feature is logging as code, which is not correct!

Reply:
I meant "inference code" instead of or in addition to "model code," not instead of "training code" — reflecting the workflow of separating training and inference code.

Clear enough without the change, regardless.